test_apps/smokeTests/smokeTests.py

```diff
@@ -209,10 +209,10 @@ def main():
         os.mkdir( resultsDir )
 
     for grid in gridSizes:
-        if ( testGrid(grid) > 0):
+        if ( testGrid(grid) != 0):
             print ("  See artifact file " + grid + ".txt or " + resultsDir + grid + ".txt for mismatches")
             print ("  Failed assertions, if any, are shown above.\n" )
-            #sys.exit(1)
+            sys.exit(-1)
```

Comment:
It's a small nit, but this should exit with a 1, not a negative 1. There is some discussion on exit codes here:

https://en.wikipedia.org/wiki/Exit_status

A little edification: the convention for unix system calls and the C standard library is  for functions to return 0 on success and a negative int (usually -1, but often not specified) on failure. This is the convention that the VAPOR API follows.

However, the *exit* codes of a Unix process are positive integers, and the convention is to return 0 on success, but a value >0 on failure. A '1' represents a generic error, other values have specified meanings.  Hence, we typically use '1' for non-normal process termination.

Lastly, there are two ways to gracefully exit a C/C++ program: calling exit() from anywhere, or returning from main(). Calling exit() will prevent C++ destructors from running, and should typically be avoided. Hence, main() should have a return statement that provides an appropriate exit code. Unfortunately, the convention when returning from main with a general error is to do 'return(1)', NOT return(-1), because exit codes are positive ints.

Reply:
Interesting.  It looks like [Python follows roughly the same convention as C++](https://docs.python.org/3/library/sys.html#sys.exit).  It looks like it tries to do cleanup as well so it might be safer to use than in C++.

I've made the fix and will push it once my sanity test finishes.

Reply:
Fixed.